File: src/test/go/test/GetUserDepartmentsTest.go
package main

import (
"authing-go-sdk/client"
"authing-go-sdk/dto"
"fmt"
)

func main() {
    options := client.ManagementClientOptions{
        AccessKeyId:     "AUTHING_USERPOOL_ID",
        AccessKeySecret: "AUTHING_USERPOOL_SECRET",
    }
    var err error
    client, err := client.NewClient(&options)
    if err != nil {
        panic(err)
    }
    request := dto.GetUserDepartmentsDto{
        UserId: "userId_8773",
        UserIdType: "userIdType_6717",
        Page: 0,
        Limit: 0,
        WithCustomData: false,
        SortBy: "sortBy_7017",
        OrderBy: "orderBy_4858",
    }
    response := client.GetUserDepartments(&request)
    fmt.Println(response)
}

File: src/test/java/test/GetUserDepartmentsTest.java
package test;
import cn.authing.sdk.java.client.ManagementClient;
import cn.authing.sdk.java.dto.*;
import cn.authing.sdk.java.model.ManagementClientOptions;
import cn.authing.sdk.java.util.JsonUtils;

import java.util.Collections;



public class GetUserDepartmentsTest {

    private static final String ACCESS_KEY_ID = "AUTHING_ACCESS_KEY_ID";
    private static final String ACCESS_KEY_SECRET = "AUTHING_ACCESS_KEY_SECRET";

    public static void main(String[] args) throws Throwable {
        ManagementClientOptions clientOptions = new ManagementClientOptions();
        clientOptions.setAccessKeyId(ACCESS_KEY_ID);
        clientOptions.setAccessKeySecret(ACCESS_KEY_SECRET);
        ManagementClient client = new ManagementClient(clientOptions);

        GetUserDepartmentsDto request = new GetUserDepartmentsDto();
        request.setUserId("userId_7446");
        request.setUserIdType("userIdType_4073");
        request.setPage(0);
        request.setLimit(0);
        request.setWithCustomData(Boolean.TRUE);
        request.setSortBy("sortBy_9427");
        request.setOrderBy("orderBy_6437");

        UserDepartmentPaginatedRespDto response = client.getUserDepartments(request);
        System.out.println(JsonUtils.serialize(response));
    }

}